dask/dataframe/io/tests/test_sql.py

```diff
@@ -332,3 +335,98 @@ def test_no_character_index_without_divisions(db):
     # attempt to read the sql table with a character index and no divisions
     with pytest.raises(TypeError):
         read_sql_table("test", db, npartitions=2, index_col="name", divisions=None)
+
+
+def test_to_sql():
+    df_by_age = df.set_index("age")
+    df_appended = pd.concat([df, df,])
+
+    # Set this to False to use a sqlite DB file under the current directory, and not delete it (for easier debugging
+    # after tests have run)
+    rm_tmpdir = True
+    if rm_tmpdir:
+
+        @contextmanager
+        def tmp():
+            with tmpfile() as f:
+                yield "sqlite:///%s" % f
+
+    else:
+        from pathlib import Path
+
+        path = Path.cwd() / "tmp" / "sqlite"
+        path.parent.mkdir(exist_ok=True, parents=True)
+        uri = f"sqlite:///{path}"
+
+        @contextmanager
+        def tmp():
+            if path.exists():
+                path.unlink()
+            yield uri
+
+    for npartitions in [1, 2, 10]:
+        ddf = dd.from_pandas(df, npartitions)
+        ddf_by_age = ddf.set_index("age")
+
+        # Simple round trip test: use existing "number" index_col
+        with tmp() as uri:
+            ddf.to_sql("test", uri)
+            actual_ddf = read_sql_table("test", uri, "number").compute()
+            assert_eq(df, actual_ddf)
+
+        # Test writing no index, and reading back in with one of the other columns as index (`read_sql_table` requires
+        # an index_col)
+        with tmp() as uri:
+            ddf.to_sql("test", uri, index=False)
+
+            actual_negish_ddf = read_sql_table("test", uri, "negish").compute()
+            assert_eq(df.set_index("negish"), actual_negish_ddf)
+
+            actual_age_ddf = read_sql_table("test", uri, "age").compute()
+            assert_eq(df_by_age, actual_age_ddf)
+
+        # Index by "age" instead
+        with tmp() as uri:
+            ddf_by_age.to_sql(
+                "test", uri,
+            )
+            actual_ddf = read_sql_table("test", uri, "age").compute()
+            assert_eq(df_by_age, actual_ddf)
+
+        # Index column can't have "object" dtype if no partitions are provided
+        with tmp() as uri:
+            ddf.set_index("name").to_sql("test", uri)
+            with pytest.raises(Exception):
+                read_sql_table("test", uri, "name").compute()
+
+        # Test various "if_exists" values
+        with tmp() as uri:
+            ddf.to_sql("test", uri)
+
+            # Writing a table that already exists fails
+            with pytest.raises(Exception):
```

Comment:
Can you assert the exact exception raised by pandas? Want to make sure it's not Dask that's doing something wrong.

Reply:
good idea, updated here and other similar `pytest.raises`

(sry, missed this the first time around!)